adobe/react-spectrum
# fix(#2444): useRestoreFocus when nodeToRestore has been unmounted

Restore focus to previous nodeToRestore when expected nodeToRestore is no longer present, as with a Menu that opens a Dialog.


Closes #2444

## ✅ Pull Request Checklist:

- [X] Included link to corresponding React Spectrum GitHub Issue #2444.

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/4b09194340f607fab1c56739c71f99e8a7d58337/storybook/index.html?path=/story/dialogcontainer--in-a-menu
2. Click or navigate to the "Open menu" button and press Enter or Space to expand the menu.
3. Click or arrow down to and activate the "Open dialog..." menu item.
4. Press Esc key to close the Dialog.
5. Focus should be restored to the "Open menu" button.

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/4b09194340f607fab1c56739c71f99e8a7d58337/storybook/index.html?path=/story/dialogcontainer--nested-dialog-containers
2. Click or navigate to the "Open menu" button and press Enter or Space to expand the menu.
3. Click or arrow down to and activate the "Do this..." or "Do that..." menu item.
4. Click or activate the "Do that" or "Do this" button to replace the Dialog with the nested Dialog.
5. Repeat step 4 a few times for good measure.
6. Press Esc key, or use the "Dismiss" button to close the Dialog.
7. Focus should be restored to the "Open menu" button.

Verify that https://reactspectrum.blob.core.windows.net/reactspectrum/4b09194340f607fab1c56739c71f99e8a7d58337/storybook/index.html?path=/story/focusscope--keyboard-navigation and https://reactspectrum.blob.core.windows.net/reactspectrum/4b09194340f607fab1c56739c71f99e8a7d58337/storybook/index.html?path=/story/focusscope--keyboard-navigation-inside-portal manage and restore focus appropriately.

## 🧢 Your Project:

Adobe/Accessibility
